cppsolutions/Dinosaurs.cpp

```diff
@@ -0,0 +1,141 @@
+#include <iostream>
+#include <vector>
+#include <string>
+#include <fstream>
+#include <algorithm>
+
+using namespace std;
+
+class DataSetStructure
+{
+public:
+	string name;
+	double length;
+	string dietOrStance;
+};
+
+class DataSet
+{
+public:
+	vector<DataSetStructure> dataSet;
+};
+
+class Dinosaurs
+{
+
+private:
+
+	DataSetStructure parseString(string line)
+	{
+		DataSetStructure ds;
+
+		if (line.size() != 0)
+		{
+			vector<char> cstr(line.c_str(), line.c_str() + line.size() + 1);
+
+			int position[2];
+			int k = 0;
+			for (int i = 0; i < cstr.size(); i++)
+			{
+				if (cstr[i] == ',')
+				{
+					position[k++] = i;
+				}
+			}
+
+			ds.name = line.substr(0, position[0]);
+			ds.length = stod(line.substr(position[0] + 1, position[1] - position[0]));
+			ds.dietOrStance = line.substr(position[1] + 1, line.size() - position[1]);
+		}
+		
+		return ds;
+	}
+
+	DataSet readCSV(string fileName)
+	{
+		ifstream dsIn;
+		dsIn.open(fileName);
+
+		DataSetStructure data;
+		DataSet ds;
+
+		string line;
+		int i = 0;
+
+		while (!dsIn.eof())
+		{
+			getline(dsIn, line, '\n');
+			if (i > 0)
+			{
+				data = parseString(line);
+				if (!data.name.empty())
+					ds.dataSet.push_back(data);
+			}
+			i++;
+		}
+		return ds;
+	}
+
+	bool doesNameExist(string name, DataSet ds)
+	{
+		bool exists = false;
+		int length = ds.dataSet.size();
+
+		for (int i = 0; i < length; i++)
+		{
+			if (ds.dataSet[i].name == name)
+			{
+				exists = true;
+				break;
+			}
+		}
+		return exists;
+	}
+
+	vector<pair<double, string>> calculateSpeedAndMergeDataSets(DataSet ds1, DataSet ds2)
+	{
+		vector<pair<double, string>> results;
+
+		int length = ds2.dataSet.size();
+		double g = 9.81;
+
+		for (int i = 0; i < length; i++)
+		{
+			string name = ds2.dataSet[i].name;
+			string stance = ds2.dataSet[i].dietOrStance;
+			double strideLength = ds2.dataSet[i].length;
+			double legLength = ds1.dataSet[i].length;
```

Comment:
Are you still expecting that the both dataset has same dinousaurs in same line number?